geps/gep-1713/index.md

```diff
@@ -0,0 +1,381 @@
+# GEP-1713: Standard Mechanism to Merge Multiple Gateways
+
+* Issue: [#1713](/kubernetes-sigs/gateway-api/issues/1713)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## tl;dr
+
+The Gateway Resource is a contention point since it is the only place to attach listeners with certificates. We propose a mechanism to allow distinct Gateway resources to be logically merged.
+
+## Goals
+
+- Define a mechanic to merge multiple Gateways (logically)
+- Define a set of acceptable properties that can be merged and their semantics
+
+## Non-Goals
+
+- Apply a Gateway resource onto N distinct gateways (one to many)
+
+## Introduction
+
+Knative generates on demand per-service certificates using HTTP-01 challenges. 
+There can be O(1000) Knative Services in the cluster which means we have O(1000) distinct certificates. 
+Thus updating a single Gateway resource with this many certificates is a contention point and inhibits horizontal scaling of our controllers.
+[Istio Ambient](https://istio.io/v1.15/blog/2022/introducing-ambient-mesh/), similarly, creates a listener per Kubernetes service.
```

Comment:
> maybe there's a simpler solution than Gateway merging?

Care to provide some alternative solutions? I'm going to close this out end of week otherwise.

Reply:
I agree there are lots of other reasons we'd want to specify gateway merging - the main one would be having fewer resources to manage.  If multiple listeners shared an endpoint, may as well merge them to reduce traffic as well.  If this proposal isn't really motivated by making things more efficient for those reasons, maybe gateway merging isn't the right solution.

Reply:
I started a doc for brainstorming here: https://docs.google.com/document/d/1qj7Xog2t2fWRuzOeTsWkabUaVeOF7_2t_7appe8EXwA/edit. I've added this proposal + a couple alternatives there, interested in any additional ideas people have. (They all include some form of Gateway merging for the record).